ush/set_cycle_and_obs_timeinfo.py

```diff
@@ -0,0 +1,736 @@
+
+        #
+        # Get the availability interval of the current observation type from the
+        # verification configuration dictionary and use it to calculate the hours-
+        # of-day at which the obs will be available.
+        #
+        # Get the obs availability interval.
+        config_var_name = "".join([obtype, "_OBS_AVAIL_INTVL_HRS"])
+        obs_avail_intvl_hrs = vx_config[config_var_name]
+        # Ensure that the obs availability interval evenly divides into 24.
+        remainder = 24 % obs_avail_intvl_hrs
+        if remainder != 0:
+            msg = dedent(f"""
+                The obs availability interval for obs of type {obtype} must divide evenly
+                into 24 but doesn't:
+                  obs_avail_intvl_hrs = {obs_avail_intvl_hrs}
+                  24 % obs_avail_intvl_hrs = {remainder}"
+                """)
+            raise Exception(msg)
+        # Assume that the obs are available at hour 0 of the day regardless
+        # of obs type.
+        obs_avail_hr_start = 0
+        obs_avail_hr_end = obs_avail_hr_start + 24
+        # Construct list of obs availability hours-of-day.
+        obs_avail_hrs_of_day = [hr for hr in range(obs_avail_hr_start, obs_avail_hr_end, obs_avail_intvl_hrs)]
+        obs_avail_hrs_of_day_str = ['%02d' % int(hr) for hr in obs_avail_hrs_of_day]
+        #
+        # Get the array of accumulation intervals for the current cumulative field. 
+        # Then loop over them to ensure that the constraints listed above are
+        # satisfied.  If for a given accumulation one or more of the constraints
+        # is not satisfied, remove that accumulation from the list of accumulations
+        # for the current field.
+        #
+        accum_intvls_array_name = "".join(["VX_", field_fcst, "_ACCUMS_HRS"])
+        accum_intvls_hrs = vx_config[accum_intvls_array_name]
+        #
+        # Loop through the accumulation intervals and check the temporal constraints
+        # listed above.
+        #
+        for accum_hrs in accum_intvls_hrs.copy():
+
+            accum_hh = f"{accum_hrs:02d}"
+            # Initialize a sub-sub-dictionary in one of the dictionaries to be returned.
+            fcst_obs_matched_times_all_cycles_cumul[field_fcst][accum_hh] = []
+            #
+            # Make sure that the accumulation interval is less than or equal to the 
+            # forecast length.
+            #
+            if accum_hrs > fcst_len_hrs:
+                msg = dedent(f"""
+                    The accumulation interval (accum_hrs) for the current cumulative forecast
+                    field (field_fcst) and corresponding observation type (obtype) is greater
+                    than the forecast length (fcst_len_hrs):
+                      field_fcst = {field_fcst}
+                      obtype = {obtype}
+                      accum_hrs = {accum_hrs}
+                      fcst_len_hrs = {fcst_len_hrs}
```

Comment:
Not sure if you knew about this shorthand in f-strings, it makes your strings shorter and ensures no confusing variable name typos
```suggestion
                      {field_fcst=}
                      {obtype=}
                      {accum_hrs=}
                      {fcst_len_hrs]}
```

Reply:
I use that when I have stand-alone print statements for debugging, but I didn't think to use them in the longer messages.  I'll try to do that from now on.  Changed everywhere I could find in this script and in `get_obs.py`.